source/extensions/retry/host/omit_host_metadata/omit_host_metadata.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/upstream/retry.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+class OmitHostsRetryPredicate : public Upstream::RetryHostPredicate {
+public:
+  explicit OmitHostsRetryPredicate(
+      const envoy::config::core::v3alpha::Metadata metadata_match_criteria)
+      : metadata_match_criteria_(metadata_match_criteria) {
+    const auto& filter_it = metadata_match_criteria_.filter_metadata().find(
+        Envoy::Config::MetadataFilters::get().ENVOY_LB);
+    if (filter_it == metadata_match_criteria_.filter_metadata().end()) {
+      throw EnvoyException("No metadata match criteria defined.");
+    }
+
+    for (auto const& it : filter_it->second.fields()) {
+      labelSet.push_back(std::make_pair(it.first, it.second));
```

Comment:
you might be able to just pass `it` to `push_back`? since its already a pair

Reply:
Done.